From 87c62d518fd5885c9330cc12a5d5a81c1a644940 Mon Sep 17 00:00:00 2001
From: Andreas Beckmann <anbe@debian.org>
Date: Wed, 1 Nov 2023 10:30:50 +0100
Subject: [PATCH] backport asm/page.h changes from 470.223.02

---
 conftest.sh | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/conftest.sh b/conftest.sh
index 47916e0..0418ec0 100755
--- a/conftest.sh
+++ b/conftest.sh
@@ -145,6 +145,7 @@ test_headers() {
     FILES_ARCH="$FILES_ARCH asm/powernv.h"
     FILES_ARCH="$FILES_ARCH asm/tlbflush.h"
     FILES_ARCH="$FILES_ARCH asm/pgtable_types.h"
+    FILES_ARCH="$FILES_ARCH asm/page.h"
 
     translate_and_find_header_files $HEADERS      $FILES
     translate_and_find_header_files $HEADERS_ARCH $FILES_ARCH
@@ -436,6 +437,9 @@ compile_test() {
             #if defined(NV_ASM_PGTABLE_TYPES_H_PRESENT)
             #include <asm/pgtable_types.h>
             #endif
+            #if defined(NV_ASM_PAGE_H_PRESENT)
+            #include <asm/page.h>
+            #endif
             #include <asm/set_memory.h>
             #else
             #include <asm/cacheflush.h>
@@ -457,6 +461,9 @@ compile_test() {
             #if defined(NV_ASM_PGTABLE_TYPES_H_PRESENT)
             #include <asm/pgtable_types.h>
             #endif
+            #if defined(NV_ASM_PAGE_H_PRESENT)
+            #include <asm/page.h>
+            #endif
             #include <asm/set_memory.h>
             #else
             #include <asm/cacheflush.h>
@@ -513,6 +520,9 @@ compile_test() {
             #if defined(NV_ASM_PGTABLE_TYPES_H_PRESENT)
             #include <asm/pgtable_types.h>
             #endif
+            #if defined(NV_ASM_PAGE_H_PRESENT)
+            #include <asm/page.h>
+            #endif
             #include <asm/set_memory.h>
             #else
             #include <asm/cacheflush.h>
-- 
2.20.1

